### ISSUE TITLE
`cast_precision_loss` is a bit misleading for `float`s

### ISSUE THREAD

**Issue by Rudxain** [rust-lang/rust-clippy#10075]
### Description

I have this code
```rust
#[warn(clippy::cast_precision_loss)]
fn div_usize_as_f64(n: usize, d: usize) -> f64 {
	n as f64 / d as f64
}
```

I get the following msg
```
warning: casting `usize` to `f64` causes a loss of precision on targets with 64-bit wide pointers (`usize` is 64 bits wide, but `f64`'s mantissa is only 52 bits wide)
```

"mantissa is 52bits" is true, but the implicit bit makes the "effective mantissa" 53b instead. I suggest making the message "more correct" by saying something like "maximum f64 value is `2u64.pow(53) - 1`"
